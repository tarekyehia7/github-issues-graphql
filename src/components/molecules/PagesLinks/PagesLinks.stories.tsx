import React from 'react';
import { withRouter } from 'storybook-addon-react-router-v6';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { PagesLinks } from './PagesLinks';

export default {
	title: 'molecules/PagesLinks',
	component: PagesLinks,
	decorators: [withRouter],
} as ComponentMeta<typeof PagesLinks>;

export const Primary: ComponentStory<typeof PagesLinks> = () => <PagesLinks />;
