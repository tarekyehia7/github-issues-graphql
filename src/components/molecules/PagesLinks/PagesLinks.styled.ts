import styled from 'styled-components';

export const PagesLinksStyled = styled.div`
	display: flex;
	justify-content: flex-start;
`;
