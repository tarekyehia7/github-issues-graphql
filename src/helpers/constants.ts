export const constants = {
    repository: 'facebook',
    projectName: 'react',
	searchTerm: 'title,body',
    issuesPerPage: 10,
    githubUrl: 'https://github.com'
};