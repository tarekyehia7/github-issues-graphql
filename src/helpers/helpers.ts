import formatDistanceToNowStrict from 'date-fns/formatDistanceToNowStrict';

export const getPagesNumber = (totalIssuesCount: number, issuesPerPage: number): number => {
	const totalPages = Math.ceil(totalIssuesCount / issuesPerPage);
	return !totalPages ? 1 : totalPages;
};

export const formatDate = (date: string) => {
	try {
		const myDate = new Date(date);
		const myFormattedDate = formatDistanceToNowStrict(myDate, { addSuffix: true });
		return myFormattedDate;
	} catch (e) {
		console.error(e);
		return '';
	}
};
