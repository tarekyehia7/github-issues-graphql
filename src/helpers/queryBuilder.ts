import { StatusEnum } from "../pages/Issues/Issues";

interface Query {
    repo: string;
    input: string;
    is: string;
    in: string;
    status: StatusEnum;
}

export const buildQuery = (query: Query) => {
    return Object.entries(query)
        .reduce((str, [p, val]) => `${str}${p}:${val} `, '')
        .replace('status:', '')
        .replace('input:', '')
        .trim();
};