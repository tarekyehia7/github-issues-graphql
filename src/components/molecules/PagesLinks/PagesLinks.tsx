import React from 'react';

import { Link, LinkType } from '../../atoms/link/Link';
import { PagesLinksStyled } from './PagesLinks.styled';

export const PagesLinks = () => {
	return (
		<PagesLinksStyled>
			<Link type={LinkType.HeaderLink} to={'/'}>
				Issues
			</Link>
		</PagesLinksStyled>
	);
};
