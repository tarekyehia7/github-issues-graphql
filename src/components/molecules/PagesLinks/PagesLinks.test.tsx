import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';

import { PagesLinks } from './PagesLinks';
import { PageWithTheme } from '../../../helpers/testing/helpers';

const ReposHeaderWithTheme = () => {
	return (
		<PageWithTheme>
			<PagesLinks />
		</PageWithTheme>
	);
};

describe('<IssueDetailsHeader />', () => {
	it('Should match snapshot', () => {
		const { container } = render(<ReposHeaderWithTheme />);
		expect(container).toMatchSnapshot();
	});

	it('Should have correct links', () => {
		const { getAllByRole } = render(<ReposHeaderWithTheme />);
		const links = getAllByRole('link');

		expect(links).toHaveLength(1);
		expect(links[0]).toHaveAttribute('href', `/`);
	});
});
