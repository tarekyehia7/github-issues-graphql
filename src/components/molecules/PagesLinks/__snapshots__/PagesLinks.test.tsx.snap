// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueDetailsHeader /> Should match snapshot 1`] = `
.c1 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  color: #24292f;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 8px;
  font-size: 14px;
  line-height: 30px;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  background-color: transparent;
  border: 0;
  border-radius: 6px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1:after {
  position: absolute;
  right: 50%;
  bottom: calc(50% - 25px);
  width: 4rem;
  height: 2px;
  content: '';
  background: #fd8c73;
  border-radius: 6px;
  -webkit-transform: translate(50%,-50%);
  -ms-transform: translate(50%,-50%);
  transform: translate(50%,-50%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      Issues
    </a>
  </div>
</div>
`;
